src/node_file.h

```diff
@@ -465,20 +465,29 @@ int MKDirpSync(uv_loop_t* loop,
 
 class FSReqWrapSync {
  public:
-  FSReqWrapSync() = default;
+  FSReqWrapSync(const char* syscall = nullptr,
```

Comment:
I think we should remove FSReqWrapSync. It brings more code and complexity than the problem it resolves.

Reply:
`FSReqWrapSync` is primarily about its destructor - like it or not, this is the C++ way to do things (RAII). Another thing this PR adds is to avoid passing `syscall` `path` and `dest` to the helper and instead put them together in an aggregate, which is also a typical thing we do in C++ land.

Reply:
I trust your judgement here. Thank you for the explanation.